cocos/ui/UIRadioButton.cpp

```diff
@@ -262,37 +269,42 @@ int RadioButtonGroup::getSelectedButtonIndex() const
     return (int) _radioButtons.getIndex(_selectedRadioButton);
 }
 
-void RadioButtonGroup::setSelectedButton(int index)
+bool RadioButtonGroup::setSelectedButton(int index)
 {
     CCASSERT(index < _radioButtons.size(), "Out of array index!");
-    setSelectedButton(_radioButtons.at(index));
+    return setSelectedButton(_radioButtons.at(index));
 }
 
-void RadioButtonGroup::setSelectedButton(RadioButton* radioButton)
+bool RadioButtonGroup::setSelectedButton(RadioButton* radioButton)
 {
-    setSelectedButtonWithoutEvent(radioButton);
-    onChangedRadioButtonSelect(_selectedRadioButton);
+    if (setSelectedButtonWithoutEvent(radioButton))
+    {
+        onChangedRadioButtonSelect(_selectedRadioButton);
+        return true;
+    }
+        
+    return false;
 }
 
-void RadioButtonGroup::setSelectedButtonWithoutEvent(int index)
+bool RadioButtonGroup::setSelectedButtonWithoutEvent(int index)
 {
-    setSelectedButtonWithoutEvent(_radioButtons.at(index));
+   return setSelectedButtonWithoutEvent(_radioButtons.at(index));
 }
 
-void RadioButtonGroup::setSelectedButtonWithoutEvent(RadioButton* radioButton)
+bool RadioButtonGroup::setSelectedButtonWithoutEvent(RadioButton* radioButton)
 {
-    if(!_allowedNoSelection && radioButton == nullptr)
+    if(!_allowedNoSelection || radioButton == nullptr)
```

Comment:
@xiaofeng11 
Why did you change `&&` to `||`? In a no-selection allowed radio button group, the given `radioButton` can be null and it must be processed.